Pt interchange duration

Theodore-Chatziioannou opened the pull request:
In a recent simulation, we noticed that the "pt interaction" activity can have duration > 0. The plan handlers are ignoring those activities (see [here](https://github.com/arup-group/elara/blob/9ff050d6de4feac5ec804c0e6b1a02ba816b1851/elara/plan_handlers.py#L542)), and as a result the start times of the next leg end up wrong (since the [activity end time](https://github.com/arup-group/elara/blob/9ff050d6de4feac5ec804c0e6b1a02ba816b1851/elara/plan_handlers.py#L548) does not get updated.

We are not sure that this "pt interaction" activity represents in these cases.

I am trying to fix the timestamp bug by checking for an "end_time" tag in the "pt interaction" activity. If there is one, then pt interaction is treated as a normal activity, separating legs/trips either side.

fredshone commented on elara/plan_handlers.py:
```diff
@@ -739,7 +739,7 @@ def process_plans(self, elem):
                         act_type = stage.get('type')
 
-                        if not act_type == 'pt interaction':
+                        if not (act_type == 'pt interaction' and (stage.get('end_time') is None)):
```
This will create new trips where a `pt interaction` is encountered with an end_time. This seems wrong.

Instead i think we need to simply allow for the fact that a pt_interaction can have a duration?

ie:
```
if stage.tag == "activity":
    # update time
   if not act_type == 'pt interaction':
       # logic for new trip
```

fredshone replied:
prior logic seems to assume duration of pt interactions is 0

Theodore-Chatziioannou replied:
changed the sequence to : 
```
if stage.tag == "activity":
   if not act_type == 'pt interaction':
       # logic for new trip
   elif stage.get('end_time'):
       # update time
   else:
      # zero pt-interaction duration
```

akay-arup commented:
Agreed with @fredshone.

Also, if we are explicitly handling cases where `pt-interaction` does or does not include an `end_time`, we should include tests for both.

Theodore-Chatziioannou commented:
you are right, the `pt interaction` activity should not separate leg of trips, and only update the timestamps. I have corrected for this, and added tests for both zero-duration pt interactions and non-zero ones.

akay-arup commented:
Looks good to me @Theodore-Chatziioannou. I don't know how much capacity we have for this at the moment. Time permitting, a larger refactor on these handlers would be a good idea. Process plans is doing quite a lot of work and the control flow is pretty difficult to follow.

Theodore-Chatziioannou commented:
I agree @akay-arup - added it as issue #221 .

divyasharma-arup commented:
Thanks @Theodore-Chatziioannou , 

I can confirm based on the new elara outputs for the `2019_baseline`, that we no longer have this behaviour where a sequence appears to have multiple possible `veh_ids`. Our sample agent: `2005001498_7_E02005584_1_2005003696`, is matched as below now to help create the person_trip_leg data module. 

![image](https://user-images.githubusercontent.com/94368005/186224784-158ebd6a-015c-4d43-bedd-ede88015bf44.png)
